configs/terraform/environments/prod/image-builder-variables.tf

```
@@ -10,6 +10,18 @@ variable "signify_prod_secret_name" {
   default     = "signify-prod-secret"
 }
 
+variable "dockerhub_oat_secret_name" {
+  type        = string
+  description = "Name of the Docker Hub OAT secret. This secret is used by image-builder's dockerhub_mirror to access Docker Hub."
+  default = "docker_sap_org_service_auth_token"
+}
+
+variable "dockerhub_username" {
+  type        = string
+  description = "Name of the Docker Hub username. This is used by image-builder's dockerhub_mirror to access Docker Hub."
+  default = "sapcom"
+}
+
```

Variables "dockerhub_oat_secret_name" and "dockerhub_username" are connected and always used together. Consider creating one variable of object type with two fields. This will stress the relation between these values, group them together and reduce number of top level variables.

Something like:
```suggestion
variable "dockerhub_auth" {
  type = object({
    username    = string
    secret_name = string
  })
  description = "Configuration for Docker Hub authentication."
  default = {
    username    = "sapcom"
    secret_name = "docker_sap_org_service_auth_token"
  }
}
```

---

Added `variable "dockerhub_credentials"` which is coupling username and secret name in one object.